render/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/render/jpql/serializer/impl/JpqlMinSerializer.kt

```diff
@@ -0,0 +1,29 @@
+package com.linecorp.kotlinjdsl.render.jpql.serializer.impl
+
+import com.linecorp.kotlinjdsl.querymodel.jpql.expression.impl.JpqlMin
+import com.linecorp.kotlinjdsl.render.RenderContext
+import com.linecorp.kotlinjdsl.render.jpql.serializer.JpqlRenderSerializer
+import com.linecorp.kotlinjdsl.render.jpql.serializer.JpqlSerializer
+import com.linecorp.kotlinjdsl.render.jpql.writer.JpqlWriter
+import kotlin.reflect.KClass
+
+class JpqlMinSerializer : JpqlSerializer<JpqlMin<*>> {
+    override fun handledType(): KClass<JpqlMin<*>> {
+        return JpqlMin::class
+    }
+
+    override fun serialize(part: JpqlMin<*>, writer: JpqlWriter, context: RenderContext) {
+        val delegate = context.getValue(JpqlRenderSerializer)
+
+        writer.write("MIN")
+        writer.write("(")
+
+        if (part.distinct) {
+            writer.write("DISTINCT")
```

Comment:
I think a white space should be added after `DISTINCT`.

Could you add a white space after DISTINCT to JpqlMinSerializer. Also, could you add it to JpqlMinSerializer? I think I made it wrong before

Reply:
I think there will be many cases that white space should be added.
I have found that `SQLWriter` have `writeKeyword` method.
What do you think about adding a `writeKeyword` method to `JpqlWriter`?

Reply:
I thought that the `writeKeyword` method might confuse the interface of `JpqlWriter`.

The `writeKeyword` method represents that keywords must have a space before and after it. So if the user calls the `write` method with keywords, it should throw an error, because the `write` method just prints a string, so it can't handle keywords.

In order for the `write` method to throw an error, it must know how to handle keywords. Then I could think about why the `write` method can't handle keywords. I could consider putting a keyword parser in the `write` method, but I felt that this was too heavy for printing.

I thought again about what I wanted for the `writeKeyword` method to do, and it was to handle a white space. So I created the `writeIfAbsent` method. It can be used like `writeIfAbsent(" ")`.

For the above reasons, I did not put the `writeKeyword` method in JpqlWriter.

In addition, the result of `writeKeyword` is `MAX ( DISTINCT table.column)`, and it doesn't look good.

> I forgot to remove the `writeKeyword` method in SqlWriter because the Native Query code was wrote a while ago and I don't plan to include Native Query in the 3.0.0 release. I'll clean up the SqlWriter on the Native Query side once this week.

Reply:
Ah, I understand :)